app/src/main/java/org/wikipedia/WikipediaApp.kt

```diff
@@ -0,0 +1,318 @@
+
+    val isOnline
+        get() = connectivityReceiver.isOnline()
+
+    val haveMainActivity
+        get() = activityLifecycleHandler.haveMainActivity()
+
+    val isAnyActivityResumed
+        get() = activityLifecycleHandler.isAnyActivityResumed
+
+    init {
+        instance = this
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+
+        WikiSite.setDefaultBaseUrl(Prefs.mediaWikiBaseUrl)
+
+        // Register here rather than in AndroidManifest.xml so that we can target Android N.
+        // https://developer.android.com/topic/performance/background-optimization.html#connectivity-action
+        registerReceiver(connectivityReceiver, IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION))
+        setupLeakCanary()
+
+        // See Javadocs and http://developer.android.com/tools/support-library/index.html#rev23-4-0
+        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true)
+
+        // This handler will catch exceptions thrown from Observables after they are disposed,
+        // or from Observables that are (deliberately or not) missing an onError handler.
+        // TODO: consider more comprehensive handling of these errors.
+        RxJavaPlugins.setErrorHandler(Functions.emptyConsumer())
+
+        mainThreadHandler = Handler(mainLooper)
+
+        var channel = ReleaseUtil.getChannel(this)
+        channel = if (channel.isBlank()) "" else " $channel"
+        userAgent = String.format("WikipediaApp/%s (Android %s; %s; %s Build/%s)%s",
+                BuildConfig.VERSION_NAME,
+                Build.VERSION.RELEASE,
+                getString(R.string.device_type),
+                Build.MODEL,
+                Build.ID,
+                channel
+        )
+
+        currentTheme = unmarshalTheme(Prefs.currentThemeId)
+        languageState = AppLanguageState(this)
+        funnelManager = FunnelManager(this)
+        sessionFunnel = SessionFunnel(this)
+
+        initTabs()
+        enableWebViewDebugging()
+        registerActivityLifecycleCallbacks(activityLifecycleHandler)
+        registerComponentCallbacks(activityLifecycleHandler)
+        NotificationCategory.createNotificationChannels(this)
+        AppShortcuts.setShortcuts(this)
+
+        // Kick the notification receiver, in case it hasn't yet been started by the system.
+        NotificationPollBroadcastReceiver.startPollTask(this)
+        InstallReferrerListener.newInstance(this)
+
+        // For good measure, explicitly call our token subscription function, in case the
+        // API failed in previous attempts.
+        WikipediaFirebaseMessagingService.updateSubscription()
+
+        EventPlatformClient.setUpStreamConfigs()
+    }
+
+    /**
+     * @return the value that should go in the Accept-Language header.
+     */
+    fun getAcceptLanguage(wiki: WikiSite?): String {
+        val wikiLang = if (wiki == null || "meta" == wiki.languageCode) "" else StringUtils.defaultString(wiki.languageCode)
```

Comment:
Should we either show `wiki.languageCode` or `wiki.languageCode.orEmpty()` instead of using `StringUtils.defaultString()`?

Reply:
yes!